osal/linux/osal_common.c

```diff
@@ -64,21 +65,25 @@ osal_task_t osal_task_create (
    (void) name;
    (void) priority;
    int ret;
-   pthread_t thread;
    pthread_attr_t attr;
-   pthread_attr_init (&attr);
-   pthread_attr_setstacksize (&attr, PTHREAD_STACK_MIN + stacksize);
+   
+   if (thread == NULL) {
+       return -1;
```

Comment:
Good to use MACROS for readability and consistency.
```
#define SUCCESS 1
#define FAILURE 0
#define ERROR -1
```

Reply:
Yes for OSAL, all such defines can be in the common header file(proposed - osal/common.h) Currently agent code has similar defines in src/csmpagent/csmp.h